### DISCUSSION TITLE
Buckets in Stable Diffusion Training

### DISCUSSION THREAD

**Discussion by clemacort** [bmaltais/kohya_ss#1516] [Q&A]
Hello, I have seen in various implementations for fine-tuning Stable Diffusion (notably Kohya) that there exists a "bucketing" feature, allowing to pass images of various aspect ratios. I have trouble understanding how these images are used in training.

From my understanding of the model, it was trained in order to generate square images of shape 512x512. The generated image might be upscaled using various algorithms like SwinIR, Lanczos, ... It may not however generate images of shape 1024x512 for example. How are these used in the fine-tuning stage (with Dreambooth or LoRA) ? What about images of shape 1024x1024 ? How are these down-scaled and will it not affect the fine-tuning ?

**Comment by jferments** [comment]
I too would be interested in some more details on how bucketing is implemented and how to best utilize it during training for good results.
